taskcluster/client.py

```diff
@@ -536,9 +536,19 @@ def _makeHttpRequest(self, method, route, payload):
 
             # Try to load JSON
             try:
+                if response.headers or headers in response:
+                    if 'Content-Type' in response.headers and \
+                            'application/json' not in response.headers['Content-Type']:
+                        return None
+                    if 'Content-Length' in response.headers and \
+                            int(response.headers['Content-Length']) == 0:
+                        return None
                 return response.json()
             except ValueError:
                 return {"response": response}
+            # Error if response has no headers
+            except AttributeError:
+                return response.json()
```

Comment:
Since we want to change JSON parsing to be something which we only do when we're expecting JSON response bodies, we should probably just return `{"response": response}` here.

Reply:
okk